### ISSUE TITLE
[BUG] crc setup not able to update the plist file for a different release

### ISSUE THREAD

**Issue by praveenkumar** [code-ready/crc#1141]
### General information

  * OS: macOS
  * Hypervisor: hyperkit
  * Did you run `crc setup` before starting it (Yes/No)? Yes

When a user use a different version of crc and runs `crc setup  --enable-experimental-features` then expected scenario is crc daemon should get updated and have info about the version which the user try to use instead it will just use the older version.

Reproduce:
- Download 1.7.0 version of CRC on fresh system (no tray and daemon)
- Run `crc setup  --enable-experimental-features`
- Check the about section in the tray ( It will show correct version of CRC and bundle)
- Delete the cluster
- Download 1.8.0 and repeat those steps and again check `About` section in the tray and it is still going to point from `1.7.0` release.

In our check we just see if plist file exists and doesn't check the path string.
```
	<array>
				<string>/Users/prkumar/Downloads/crc-1.7.0/crc</string>
				<string>daemon</string>
				<string>--log-level</string>
				<string>debug</string>
	</array>
```

**Comment by gbraad** [comment]
Likely the executable was in use and therefore you see the old version. The setup should detect a running instance, stop it and then replace before starting it again. (this was mentioned as a scenario in on of the comments on a PR).

**Comment by anjannath** [comment]
We do that for the tray, as in we check if the version running is the same as the one requested, if not then we restart. In case of the daemon am thinking to always create the plist file with the current executable's path and restart it whenever setup is performed.

**Comment by gbraad** [comment]
> always create the plist file with the current executable's path 

this would be missing... but best would be to 'stop' first, as you do not want the executable to remain in memory or have a filelock due to it being active.

**Comment by praveenkumar** [comment]
fixed by #1142
